src/Foundation/NSUrl.cs

```diff
@@ -142,6 +142,21 @@ public bool SetResource (NSString nsUrlResourceKey, NSObject value)
 			return SetResourceValue (value, nsUrlResourceKey, out error);
 		}
 
+		public static bool operator ==(NSUrl x, NSUrl y)
+		{
+			if (System.Object.ReferenceEquals (x, y)) // If both are null, or both are same instance, return true.
```

Comment:
This can be written as:

    if ((object) x == (object) y)

which is smaller IL and at least as fast as ReferenceEquals (faster unless the JIT/AOT compiler does inlining).